opm/autodiff/FullyImplicitBlackoilSolver.hpp

```diff
@@ -194,6 +194,12 @@ namespace Opm {
         std::vector<int>         primalVariable_;
 
         // Private methods.
+
+        // return true if wells are available
+        bool wellsActive() const { return wells_ ? wells_->number_of_wells > 0 : false ; }
+        // return wells object
+        const Wells& wells () const { assert( wells_ ); return *wells_; }
```

Comment:
The statement
```C++
assert (wells_);
```
is, strictly speaking, undefined.  The input to `assert` must be an expression of scalar type with a boolean result, so you really need to write
```C++
assert (wells_ != 0);
```
instead.